scaaml/io/resume_kti.py

```diff
@@ -232,22 +238,23 @@ def __next__(self):
         calls without a _save_progress call), thus updates content of
         self._progress_filename file.
 
-        Returns:
-            Key-text pair.
+        Returns: Encryption parameters as a namedtuple.
         """
         if self._index % self._shard_length == 0:
             self._save_progress()
         if self._index >= get_any_value_len(self._parameters):
             raise StopIteration
         self._index += 1
-        return {
-            parameter_name: self._parameters[parameter_name][self._index - 1]
-            for parameter_name in self._parameters  # pylint: disable=C0206
-        }
-
-    def __len__(self):
-        """How many pairs are there in total (including the initial_index
-        skipped pairs)."""
+        return self._element_class(
+            **{
+                parameter_name: self._parameters[parameter_name][self._index -
+                                                                 1]
+                for parameter_name in self._parameters  # pylint: disable=C0206
```

Comment:
This comprehension isn't great to read, especially because of lines 250 and 251 being split.